components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/AuthzUtil.java

```diff
@@ -180,15 +181,23 @@ private static String getAccessingTenantDomain(AuthenticatedUser authenticatedUs
     }
 
     /**
-     * Get accessing user id of authenticated user.
+     * Get user ID corresponds to the accessing user ID of authenticated user.
      *
      * @param authenticatedUser Authenticated user.
-     * @return Accessing tenant domain.
-     * @throws IdentityOAuth2Exception if an error occurs while retrieving accessing user id.
+     * @return The user ID.
+     * @throws IdentityOAuth2Exception If an error occurs while resolving the user ID.
      */
-    private static String getAccessingUserId(AuthenticatedUser authenticatedUser) throws IdentityOAuth2Exception {
+    private static String getUserIdOfAssociatedUser(AuthenticatedUser authenticatedUser)
+            throws IdentityOAuth2Exception {
 
         String associatedUserId = getUserId(authenticatedUser);
```

Comment:
this can be moved as the else block

Reply:
Fixed by - https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2246/commits/0c6f09e4a6e9d351d13f6f780f7863271ba4dcfb